devto-draft.md

```diff
@@ -1 +1,393 @@
+
+Methods are like any function in javascript. The return statement is totally optional, you can just use it to mutate data in the background _(not a good practice)_
+
+## Asynchronous Gotchas
+
+This is a gotcha that you find out later in Vue, and in React.
+In React, there is a built-in method called `setState()`. When `setState()` gets called, it gets handled asynchronously in the background.
+
+Vue is no different, you have a method called `Vue.nextTick()` that awaits the latest updated value to the data.
+
+One lesson I learned is that it's best to set local variables within your methods, and only mutate data attributes as needed.
+
+## Templates
+
+With Vue, there is something called `templates`. In your component, you have the ability to declare `<template>` slots. When that component gets called, you can inject HTML where those slots match.
+
+This has been immensely useful when working in libraries. I could inject functionality as I saw fit, to get the feature I needed.
+
+Templates do not get rendered as a `<div>` element in the DOM. React has a similar approach to templates called fragments, but the approach is limited to just the first element in your component.
+
+## Lifecycle methods
+
+Lifecycle methods describe what components do during their lifetime.
+
+Here are common ones you'll use in Vue:
+
+- mounted() - Similar to React's componentDidMount. This generally when making an Axios/fetch call to initialize data in your component from a backend.
+- updated() - Similar to React's componentDidUpdate. When you update data through your components, you want to send a POST request to your backend to keep things in sync
+- destroyed() - Similar to React's componentWillUnmount. You don't need to use this, but it helps clean up leftover event listeners.
+
+## Hooks
+
+Hooks makes React an extremely powerful framework. No longer did you have to use redux for handling state, you could just use `useState()` instead. No longer did you have to deal with `setState()`, there were a slew of other new improved methods to use.
+
+Vue has no equivalent to hooks, although there is an RFC at the time of writing this
+
+## Directives
+
+Vue's built-in directives make development a breeze. You could create iterable directives right inside your HTML.
+
+In React, this often meant making another component called "group(s)" and maybe one called "group".
+
+Vue, you can just dump the `v-for`, `v-if` etc right inside your HTML. Your logic coherently just makes sense looking at it.
+
+Alternatively, you can write your own directives! This lets you attach functionality to page if necessary, making it easier to develop faster
+
+One great use case for startups is to develop a reporting system. If a user clicks on a feature that doesn't yet exist, you can add a popup modal saying "Sorry we're working hard on this new feature! Come back later". In the background, you can make Axios request to notify users really do want that feature.
```

Comment:
How is this a showcase of directive? I suggest unpacking more how this is an example of a directive and perhaps show a code example.